### PR TITLE
EP-4799: Fix handover database infos regex

### PR DESCRIPTION
**Pull request by luca-drf**

### CODE REVIEW

File: ensembl_prodinf/handover_tasks.py [hunk 1]
```diff
@@ -109,19 +109,19 @@ def check_db(uri):
...
+        release = m.group('release')
+        assembly = m.group('assembly')
         return db_prefix, db_type, release, assembly
     elif compara_pattern.match(database):
-        db_type = compara_pattern.match(database).group(1)
-        db_prefix = compara_pattern.match(database).group(2)
-        return db_prefix, db_type, None, None
+        m = compara_pattern.match(database)
+        division = m.group('division')
+        db_prefix = division if division else 'compara'
```
Comment (thomasmaurel) [inline ensembl_prodinf/handover_tasks.py]: For vertebrates, the current code expects db_prefix  not to be set: https://github.com/Ensembl/ensembl-prodinf-core/blob/release/2.1/ensembl_prodinf/handover_tasks.py#L154
I think there is two solutions, either we leave db_prefix unset for vertebrates or set this to 'ensembl' and then remove the else bit from above.

Comment (luca-drf) [reply ensembl_prodinf/handover_tasks.py]: My mistake. However, I think that if `division` is `None` we should pass `''` and not modify the _else_ block.

Comment (luca-drf) [reply ensembl_prodinf/handover_tasks.py]: Fixed in d2c252a

**Review by thomasmaurel — APPROVED** [review]
Looks good. Much cleaner code than before.

**Pull request merged**